crates/ruff_python_formatter/src/module/mod_module.rs

```diff
@@ -1,18 +1,19 @@
-use crate::AsFormat;
+use crate::builders::PyFormatterExtensions;
+use crate::context::NodeLevel;
 use crate::{FormatNodeRule, PyFormatter};
 use ruff_formatter::prelude::hard_line_break;
-use ruff_formatter::{write, Buffer, FormatResult};
-
+use ruff_formatter::{Format, FormatResult};
 use rustpython_parser::ast::ModModule;
 
 #[derive(Default)]
 pub struct FormatModModule;
 
 impl FormatNodeRule<ModModule> for FormatModModule {
     fn fmt_fields(&self, item: &ModModule, f: &mut PyFormatter) -> FormatResult<()> {
-        for stmt in &item.body {
-            write!(f, [stmt.format(), hard_line_break()])?;
-        }
+        f.join_nodes(NodeLevel::TopLevel)
+            .nodes(&item.body)
+            .finish()?;
```

Comment:
Should this use `item.body.format().with_options(SuiteLevel::TopLevel)` (The `Suite` formatting) so that it inserts the right amount of new lines before and after classes and functions?

Reply:
done